runtime/compiler/env/VMJ9Server.cpp

```diff
@@ -343,20 +344,61 @@ TR_J9ServerVM::getSystemClassLoader()
    }
 
 bool
-TR_J9ServerVM::jitFieldsAreSame(TR_ResolvedMethod * method1, I_32 cpIndex1, TR_ResolvedMethod * method2, I_32 cpIndex2, int32_t isStatic)
+TR_J9ServerVM::jitFieldsOrStaticsAreIdentical(TR_ResolvedMethod * method1, I_32 cpIndex1, TR_ResolvedMethod * method2, I_32 cpIndex2, int32_t isStatic)
    {
-   JITServer::ServerStream *stream = _compInfoPT->getMethodBeingCompiled()->_stream;
-
-   // Pass pointers to client mirrors of the ResolvedMethod objects instead of local objects
-   TR_ResolvedJ9JITServerMethod *serverMethod1 = static_cast<TR_ResolvedJ9JITServerMethod*>(method1);
-   TR_ResolvedJ9JITServerMethod *serverMethod2 = static_cast<TR_ResolvedJ9JITServerMethod*>(method2);
-   TR_ResolvedMethod *clientMethod1 = serverMethod1->getRemoteMirror();
-   TR_ResolvedMethod *clientMethod2 = serverMethod2->getRemoteMirror();
+   auto serverMethod1 = static_cast<TR_ResolvedJ9JITServerMethod*>(method1);
+   auto serverMethod2 = static_cast<TR_ResolvedJ9JITServerMethod*>(method2);
+   J9Class *ramClass1 = serverMethod1->constantPoolHdr();
+   J9Class *ramClass2 = serverMethod2->constantPoolHdr();
+   UDATA field1 = 0, field2 = 0;
+   J9Class *declaringClass1 = NULL, *declaringClass2 = NULL;
 
    bool result = false;
+   bool needRemoteCall = true;
+   bool cached = getCachedField(ramClass1, cpIndex1, &declaringClass1, &field1);
```

Comment:
I would prefer a `logical and` rather than a `bit-wise and` operator here:
```
bool cached = getCachedField(ramClass1, cpIndex1, &declaringClass1, &field1) &&
              getCachedField(ramClass2, cpIndex2, &declaringClass2, &field2);
```

Reply:
Done